contributor-docs/adrs/adr-008-experimental-components.md

```diff
@@ -0,0 +1,75 @@
+
+1. **Need for experimental candidates**: As more teams build React components and look to upstream reusable components, it is natural to look at drafts as the place to put them. However, the barrier to entry into drafts isn't low (because we still follow design reviews, dev conventions, API reviews, etc.)
+
+   Feature teams using Primer may not be intimately familiar with Primer component lifecycle criteria or have the time and/or knowledge to meet it. Often, feature teams create their own components outside of Primer if they find it too difficult to contribute. This leads to lower Primer adoption and conflicting versions of similar functionality throughout GitHub products.
+
+   We want to enable feature teams to contribute to Primer. We also want to make it clear which contributions haven't been evaluated by our high standards for Primer components.
+
+
+
+&nbsp;
+
+## Decision
+
+1. npm package: "Experimental" components, as defined by the component lifecycle, should not be part of semantically versioned npm package for `@primer/react`.
+
+2. npm package: Each experimental component should be independently published and versioned with semver. Breaking changes should be tagged as major versions. Example: `@github/react-commentbox`
+
+3. The code for experimental components should live in [github/github/modules/react-shared](https://github.com/github/github/tree/master/app/assets/modules/react-shared) and published as individual packages as suggested in point 2.
+
+   The monolith already has a place where reusable react components live. This has the lowest barrier to entry for folks working in the monolith.
+
+#### Risks:
+
+This decision would increase the amount of work required for non-monolith projects when upstreaming reusable components. Without improving the development and publishing workflow for react-shared, this could create more friction and make contributions more difficult.
+
+
+&nbsp;
+
+#### Other options considered
+
+1. The code for experimental components should live in a new repository code `github.com/primer/react-candidates` to support different conventions and processes for experimental components and convey shared ownership between primer and product teams. 
+ 
+   Keeping experimental components in primer _org_ suggests that the primer _team_ would take up maintenance of these components while they are still candidates (bugs, a11y remedial, etc.). This will be a new parallel workstream for the team and with our current team size, we might not be able to give it the required attention.
+
+2. The code for experimental components should live in a new repository `github/react-shared` or `github/primer-react-candidates` instead of `primer/react-candidates`
```

Comment:
Note: This is a very close second and we should keep this in option in mind

Cons: 
1. More work for folks working on the monolith to contribute to an external package

Pros:
1. A lot more accessible for non-monolith projects to upstream to this (vs the react-shared in monolith)
2. Cleaner versioning and dependency management for both monolith and non-monolith projects, easier to avoid mistakes.
3. Easier to make tooling, publishing workflow and documentation primer-adjacent (compared to the monolith)

Reply:
> More work for folks working on the monolith to contribute to an external package

Could you elaborate? There is the obvious "would require NPM linking to integrate and test", but that could be mitigated by having proper testing (e.g. Storybook) in that repo, which would allow for rapid testing and prototyping.

From a contribution standpoint, I would actually say it'd be easier. Assuming the bar for experimental components would be lower than that of Primer proper, it would allow for teams to contribute components and rapidly iterate on them.

There is, however, an un-answered question (possibly what you're referring to): How do we manage versions in the monolith? Specifically, if the Issues App team needs `Component@3.0.0` and PRs needs `Component@4.0.0`, how can we make sure that these components are able to live alongside each other? For breaking changes, this could cause major issues in the monolith and shifts the burden of responsibility on the person or team that is upgrading the component to _also_ upgrade all other instances that might be broken by the new version if there is a single version available in the monolith.